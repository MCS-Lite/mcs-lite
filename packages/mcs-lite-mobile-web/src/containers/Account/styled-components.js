import styled from 'styled-components';
import Button from 'mcs-lite-ui/lib/Button';
import P from 'mcs-lite-ui/lib/P';
import Logo from '../../components/Logo';

export const FlatButton = styled(Button)`
  border: initial;
  border-radius: initial;
  height: 56px;
  border-top: 1px solid ${props => props.theme.color.white};
`;

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  flex-grow: 1;
`;

export const Body = styled.div`
  width: 100%;
  flex-grow: 1;
  display: flex;
  align-items: center;
  flex-direction: column;

  > *:not(:first-child) {
    margin-bottom: 8px;
  }
`;

export const StyledLogo = styled(Logo)`
  margin-bottom: 32px;
`;

export const VersionWrapper = styled(P)`
  text-align: center;
  margin-bottom: 8px;
`;

export const Footer = styled.footer`
  width: 100%;
`;
