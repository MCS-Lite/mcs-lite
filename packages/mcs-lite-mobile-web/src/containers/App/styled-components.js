import styled from 'styled-components';
import { MobileContentWrapper } from 'mcs-lite-ui';

export const HeightContainer = styled.div`
  height: 100%;
`;

export const Fixed = styled.div`
  position: fixed;
  bottom: 56px;
  width: 100%;
`;

export const ToastContainer = styled(MobileContentWrapper)`
  padding: 16px;

  > * {
    margin-top: 8px;
  }
`;
