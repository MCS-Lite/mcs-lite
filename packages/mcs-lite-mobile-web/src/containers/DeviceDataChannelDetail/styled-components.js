import styled from 'styled-components';
import { MobileContentWrapper, Small, A } from 'mcs-lite-ui';

export const CardContainer = styled(MobileContentWrapper)`
  padding: 8px;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayBase};
`;

export const HistoryContainer = styled(MobileContentWrapper)`
  padding: 16px;
`;

export const HistoryHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 8px;
`;

export const ResetWrapper = styled(A)`
  display: flex;
  font-size: 24px;
  align-items: center;

  > *:first-child {
    margin-right: 4px;
  }
`;
