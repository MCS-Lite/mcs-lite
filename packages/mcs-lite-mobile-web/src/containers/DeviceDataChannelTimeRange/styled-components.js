import styled from 'styled-components';
import MobileContentWrapper from 'mcs-lite-ui/lib/MobileContentWrapper';

export const Container = styled(MobileContentWrapper)`
  padding: 16px 16px 56px 16px;
`;

export const InputWrapper = styled.div`
  margin-bottom: 8px;

  > *:first-child {
    margin-bottom: 4px;
  }
`;

export const FakeInput = styled.div`
  display: flex;
  align-items: center;
  box-sizing: border-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 3px;
  padding: 0 10px;
  cursor: pointer;
  min-height: ${props => props.theme.height.normal};
  color: ${props => props.theme.color.black};
  background-color: ${props => props.theme.color.white};
  border-color: ${props => props.theme.color.grayDark};
  font-size: ${props => props.theme.fontSize.p};
  flex-shrink: 0;
  flex: 1;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  padding: 8px 16px;

  > * {
    flex: 1;
  }

  > *:first-child {
    margin-right: 8px;
  }
`;
