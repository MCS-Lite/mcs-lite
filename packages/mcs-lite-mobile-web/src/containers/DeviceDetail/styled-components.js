import styled from 'styled-components';
import { Img, MobileContentWrapper } from 'mcs-lite-ui';
import { Link } from 'react-router';

export const Container = styled(MobileContentWrapper)`
  padding: 4px;
`;

export const StyledImg = styled(Img)`
  height: 192px;
`;

export const CardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;

  > * {
    height: initial;
    padding: 8px 16px;
    margin: 4px;
    flex-basis: 100%;
  }

  > [data-width ~= half] {
    flex-grow: 1;
    flex-basis: 40%;
  }
`;


export const StyledLink = styled(Link)`
  display: block;
  text-decoration: none;
  cursor: pointer;
  color: currentColor;
`;
