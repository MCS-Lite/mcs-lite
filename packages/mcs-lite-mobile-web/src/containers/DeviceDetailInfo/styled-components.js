/* eslint import/prefer-default-export: 0 */

import styled from 'styled-components';
import MobileContentWrapper from 'mcs-lite-ui/lib/MobileContentWrapper';

export const Container = styled(MobileContentWrapper)`
  padding: 16px 16px 56px 16px;

  > div + div {
    margin-top: 8px;
  }

  p {
    word-wrap: break-word;
  }
`;
