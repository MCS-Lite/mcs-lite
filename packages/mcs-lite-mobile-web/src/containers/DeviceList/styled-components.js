import React from 'react';
import styled from 'styled-components';
import R from 'ramda';
import { P, MobileHeader, MobileContentWrapper } from 'mcs-lite-ui';
import { opacity } from 'mcs-lite-theme';

export const Container = styled(MobileContentWrapper)`
  padding: 8px;
`;

export const CardWrapper = styled.div`
  > * {
    margin-bottom: 8px;
  }
`;

const omitProps = R.omit(['active']);
const BaseComponent = props => <MobileHeader.MobileHeaderIcon {...omitProps(props)} />;
export const StyledHeaderIcon = styled(BaseComponent)`
  color: ${props => opacity(props.active ? 0.5 : 1)(props.theme.color.white)};
`;

export const PlaceholdWrapper = styled(P)`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  width: 100%;
  left: 0;
  top: ${props => props.theme.mobile.headerHeight};
  bottom: 0;
`;
