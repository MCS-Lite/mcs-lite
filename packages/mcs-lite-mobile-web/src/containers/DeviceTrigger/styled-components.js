import styled from 'styled-components';
import Small from 'mcs-lite-ui/lib/Small';
import MobileContentWrapper from 'mcs-lite-ui/lib/MobileContentWrapper';

export const Item = styled(MobileContentWrapper)`
  padding: 8px 16px;
  border-bottom: 1px solid ${props => props.theme.color.grayBase};
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayDark};
`;

export const IconWrapper = styled.div`
  font-size: 24px;
  color: ${props => props.theme.color.primary};
`;
