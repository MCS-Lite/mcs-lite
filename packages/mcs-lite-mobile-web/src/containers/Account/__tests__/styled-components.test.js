import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import mobileTheme from '../../../utils/mobileTheme';
import {
  FlatButton,
  Container,
  Body,
  StyledLogo,
  Footer,
} from '../styled-components';

jest.mock('mcs-lite-ui/lib/Button');
jest.mock('mcs-lite-ui/lib/P');

it('should render components correctly', () => {
  const wrapper = mount(
    <ThemeProvider theme={mobileTheme}>
      <div>
        <FlatButton />
        <Container />
        <Body />
        <StyledLogo />
        <Footer />
      </div>
    </ThemeProvider>,
  );

  expect(toJson(wrapper.find('div'))).toMatchSnapshot();
});
