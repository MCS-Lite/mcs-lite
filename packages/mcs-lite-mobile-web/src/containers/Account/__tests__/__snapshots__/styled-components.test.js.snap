exports[`test should render components correctly 1`] = `
Array [
  <div>
    <Styled(Button)>
        <Button
            block={false}
            className="buRwqu"
            component="button"
            kind="primary" />
    </Styled(Button)>
    <styled.div>
        <div
            className="cCgbat" />
    </styled.div>
    <styled.div>
        <div
            className="EuRNj" />
    </styled.div>
    <styled.img
        alt="MCS Lite Logo"
        src="logo.svg">
        <img
            alt="MCS Lite Logo"
            className="fRRFQD"
            src="logo.svg" />
    </styled.img>
    <styled.footer>
        <footer
            className="bMAZFf" />
    </styled.footer>
</div>,
  <div
    className="cCgbat" />,
  <div
    className="EuRNj" />,
]
`;
